src/main/kotlin/controllers/CacheChooserController.kt

```diff
@@ -281,53 +306,169 @@ class CacheChooserController(
+                            } else {
+                                entryName
+                            }
+                            val dest = File(destFolder, "cache/$relativePath")
+                            dest.parentFile?.mkdirs()
+                            // Use REPLACE_EXISTING to overwrite files if they exist
+                            Files.copy(zipIn, dest.toPath(), StandardCopyOption.REPLACE_EXISTING)
+                        }
+                        zipIn.closeEntry()
+                        zipEntry = zipIn.nextEntry
+                    }
+                    zipIn.close()
+
+                    // Also download the keys if available
+                    try {
+                        val openRs2Api = OpenRs2Api()
+                        val keys = openRs2Api.getCacheKeysById(cache.scope, cache.id.toString())
+                        if (keys != null && keys.isNotEmpty()) {
+                            val filePath = destFolder.toPath().resolve("xteas.json")
+                            Files.createDirectories(filePath.parent)
+                            Files.newBufferedWriter(
+                                filePath,
+                                StandardOpenOption.CREATE,
+                                StandardOpenOption.TRUNCATE_EXISTING
+                            ).use { writer ->
+                                ObjectMapper().writerWithDefaultPrettyPrinter().writeValue(writer, keys)
+                            }
+                        }
+                    } catch (e: Exception) {
+                        println("Warning: Could not download cache keys: ${e.message}")
+                        // Continue anyway, keys are optional
+                    }
+
+                    // Generate a basic params.txt file with the revision number from build
+                    try {
+                        val revisionNumber = if (cache.builds.isNotEmpty()) {
+                            cache.builds.first().major
                         } else {
-                            Files.copy(tarIn, dest.toPath())
+                            null
+                        }
+                        
+                        if (revisionNumber != null) {
+                            val paramsFile = File(destFolder, "params.txt")
+                            Files.createDirectories(destFolder.toPath())
+                            Files.newBufferedWriter(
+                                paramsFile.toPath(),
+                                StandardOpenOption.CREATE,
+                                StandardOpenOption.TRUNCATE_EXISTING
+                            ).use { writer ->
+                                writer.write("Created at ${java.time.LocalDateTime.now()}\n")
+                                writer.write("\n")
+                                writer.write("codebase=http://oldschool${revisionNumber}.runescape.com/\n")
```

Comment:
why actively fake a potentially incorrect codebase if we don't use the codebase?

Reply:
Cut down to essentials + comment saying it's not a "real" params file